compiler/ballerina-lang/src/main/java/org/ballerinalang/model/tree/SortableTypeDefNode.java

```diff
@@ -0,0 +1,29 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.model.tree;
+
+/**
+ * Interface representing type definitions that are sortable using their dependencies to each other.
+ *
+ * @since 2.0
+ */
+public interface SortableTypeDefNode extends Node {
```

Comment:
How about `OrderedNode` or `OrderedTypeDefNode`? Suggested the former since this precedence APIs seem generic enough that they can be used for other nodes as well, if needed

Reply:
+1